SU-HSDO/suhumsci
# STN-607: Implementation of Testimonial Paragraph/UI Pattern for views

# [STN-607](https://sparkbox.atlassian.net/browse/STN-607) READY FOR REVIEW

## Summary
Creates a new Testimonial Paragraph and UI Pattern to be used in views . Used on both Colorful and Traditional. The styles are different to complement each theme. 

Design comps for testimonial: [Figma](https://www.figma.com/file/RiJgL9DltLHzig4ynM34iU/H-and-S-Testimonial-Block-Quote?node-id=0%3A1)

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. In order for the testimonial pattern to be recognized, you will need to clear your cache by running `lando drush @sparkbox_sandbox.local cr`.
3. To test a paragraph type, create a new flexible page or open an existing one. Under `Components` add a `Testimonial`. Fill in all the fields. 
4. To test the testimonial as a UI pattern in a view, add a view to a flexible page. Under  `Components` add a `View`. I chose to add `Custom Econ Events` with a Display of `Past Events.`
    
    <img width="840" alt="add view" src="https://user-images.githubusercontent.com/12678977/106809409-a3128100-6639-11eb-8ec5-331ab13ec0dc.png">
<br>
    Edit the view to use the testimonial UI pattern. Under `Format > Show` click on `Settings`.
<br>
    <img width="1305" alt="veiw settings" src="https://user-images.githubusercontent.com/12678977/106809831-27650400-663a-11eb-8567-0e239bf943b6.png">
<br>
    Edit the Row style options. Set `Pattern` to `Testimonial`. Leave the `Variant` to `Top (Default)`. Set `Content: Title` to `Position`, `Content: Image` to `Image`, `Content: Body` to `Quote`, `Content: Speaker` to `Name`, and `Content: Link to Content` to `Link`. All the other sources should be set to `- Hidden -`. Apply and save the view.
<br>

![row style options](https://user-images.githubusercontent.com/12678977/106810449-f20ce600-663a-11eb-8083-594f0c385565.png)


5. Test top / bottom quote options
6. Use the Testing Checklist below as a reference for all the things you should consider testing.

## Testing Checklist
### Themes
- [x] Colorful (Note that the image on the Colorful theme has a border-radius that crops it into a circle, while the image in the Traditional theme is a square.)
- [x] Traditional

### Theme Settings
- [x] Dark Color Inversion

### Utility Classes
Layout Builder
- [x] hb-dark-pattern
- [x] hb-blockquote-black-text (When added to layout builder, this class should make the color of the quote text black.)

#### Views
- [x] hb-dark-pattern

#### Fields --> The utility classes work but I don't recommend using them on the testimonial.
- [x] hb-title
- [x] hb-highlighted-label
- [x] hb-subtitle
- [x] hb-divider

### Component considerations
- [x] Rows
- [x] Wells

### Content considerations
- [x] Rows with single and double wide content
- [x] Longer names / positions
- [x] Examples without all the fields used & different combinations of fields used.
- [x] Text area / Testimonial / Text area

## Screenshot

![sparkbox-sandbox suhumsci loc_qa_testimonial-paragraphui-pattern-views](https://user-images.githubusercontent.com/12678977/106931300-6e0f3880-66e4-11eb-8e92-3957899f22c2.png)
